.github/workflows/deploy-dev.yml

```diff
@@ -0,0 +1,185 @@
+name: Deploy Dev Enviroment
+on:
+#  schedule:
+#    - cron: '0 * * * *' # Runs every hour #TODO: Figure out cron
+  push:
+    branches:
+      - main
+    paths:
+      - 'env-dev.yml'
+  workflow_dispatch: # Allows manual triggering
+jobs:
+  check-for-payu-updates:
+    runs-on: ubuntu-latest
+    outputs:
+      commits-count: ${{ steps.check-payu-commits.outputs.commits-count }}
+      last-commit-hash: ${{ steps.check-payu-commits.outputs.latest-commit-hash }}
+    steps:
+      - name: Checkout current repository
+        uses: actions/checkout@v4
+
+      - name: Get last workflow run time
+        id: last-run-time
+        env:
+          GH_TOKEN: ${{ secrets.GH_TOKEN }}
+        run: |
+          # Fetch the last successful workflow run time
+          response=$(gh api \
+            -H "Accept: application/vnd.github+json"   \
+            -H "X-GitHub-Api-Version: 2022-11-28"   \
+            /repos/access-nri/payu-condaenv/actions/workflows/deploy-dev.yml/runs?status=success&per_page=1)
+          last_run_time=$(echo $response | jq -r '.workflow_runs[0].updated_at')
+
+          echo "Last successful workflow run time: $last_run_time"
+          echo "last-run-time=$last_run_time" >> $GITHUB_OUTPUT
+
+      - name: Checkout payu repository
+        uses: actions/checkout@v4
+        with:
+          repository: payu-org/payu
+          path: payu
+          ref: master
+
+      - name: Check commits in payu repository
+        id: check-payu-commits
+        run: |
+          # Check for any commits since last successful runtime
+          last_run_time="${{ steps.last-run-time.outputs.last-run-time }}"
+          commits_count=$(git -C ./payu rev-list --count --since="$last_run_time" master)
+
+          # Get latest commit hash
+          latest_commit_hash=$(git -C ./payu rev-parse --short HEAD)
+
+          echo "Number of new commits since last run: $commits_count, latest commit hash: $latest_commit_hash"
+
+          echo "commits-count=$commits_count" >> $GITHUB_OUTPUT
+          echo "latest-commit-hash=$latest_commit_hash" >> $GITHUB_OUTPUT
+
+  pack:
+    name: Pack Payu
+    runs-on: ubuntu-latest
+    needs:
+      - check-for-payu-updates
+    # Deploy payu if manually triggered, env-dev.yml has been updated, or if there's new commits to payu repository
+    if: >
+      needs.check-for-payu-updates.outputs.commits-count > 0 ||
+      github.event_name == 'push' ||
+      github.event_name == 'workflow_dispatch'
```

Comment:
Haven't seen the `>` syntax before! Very cool :) 
Currently this will always be successful as it is triggered by `on.push` or `on.workflow_dispatch` (the `on.schedule` is commented out). Is `on.schedule` being added later?

Reply:
Yeah I wasn't sure on how often to schedule it. I've added in a cron to run at 7AM AEST. If it needs to run more often, could edit it at a later point.